### PR TITLE
:seedling: Added authorize with session Token method

### PR DESCRIPTION
**Pull request by igormartsekha**

### CODE REVIEW

File: app/src/main/java/com/okta/appauth/android/example/SessionAuthorizeActivity.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+    /**
+     * Starts an authorization flow with the OktaAppAuth object. Make sure that the object is
+     * initialized before this is called.
+     */
+    @MainThread
+    private void startAuth() {
+        displayLoading(getString(R.string.loading_authorizing));
+
+        String sessionToken = ((EditText)findViewById(R.id.session_token_value))
```
Comment (smokienko) [inline app/src/main/java/com/okta/appauth/android/example/SessionAuthorizeActivity.java]: Please leave a comment that will describe how to get sessionToken.

File: library/src/main/java/com/okta/appauth/android/DirectAuthorizationService.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
+
+class DirectAuthorizationService {
```
Comment (smokienko) [inline library/src/main/java/com/okta/appauth/android/DirectAuthorizationService.java]: Let's name it SesssionAuthorizationService.

File: library/src/test/java/com/okta/ConnectionBuilderForTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+/**
+ * An example implementation of {@link ConnectionBuilder} that permits connecting to http
+ * links, and ignores certificates for https connections. *THIS SHOULD NOT BE USED IN PRODUCTION
+ * CODE*. It is intended to facilitate easier testing of AppAuth against development servers
+ * only.
+ */
+public class ConnectionBuilderForTest implements ConnectionBuilder {
```
Comment (bdemers) [inline library/src/test/java/com/okta/ConnectionBuilderForTest.java]: Can you explain why we have this? Is this for a mock test server or something?

**Comment by bdemers** [comment]
Is it possible to break this up at all?

File: library/src/test/java/com/okta/ConnectionBuilderForTest.java (hunk 3 above)
Comment (ihormartsekha-okta) [reply library/src/test/java/com/okta/ConnectionBuilderForTest.java]: Yes, this is for MockWebServer. We emulate HTTP server to test our network requests

File: library/src/main/java/com/okta/appauth/android/OktaAppAuth.java [hunk 4]
```diff
@@ -295,6 +294,23 @@ public void run() {
         });
     }
 
+    /**
+     * Authorize using Session Token.
+     *
+     * @param sessionToken     Session Token
+     * @param listener         The OktaAuthListener to receive callback with results
+     */
+    public void authorize(
```
Comment (nbarbettini) [inline library/src/main/java/com/okta/appauth/android/OktaAppAuth.java]: Let's name this `authenticate` to match iOS: https://github.com/okta/okta-sdk-appauth-ios/pull/113#discussion_r250386206

Comment (igormartsekha) [reply library/src/main/java/com/okta/appauth/android/OktaAppAuth.java]: Done

**Pull request merged**